root/hist/hist/THistFillBenchmarks.cxx

```diff
@@ -16,69 +15,115 @@ static void FillArguments(benchmark::internal::Benchmark *b)
          b->Args({i, j, j / 2});
 }
 
+// Create array for x coordinates
+std::vector<double> MakeXCoorVec(int nbOfDataPoints) 
+{
+   std::vector<double> vec;
+   for (int j = 0; j < nbOfDataPoints; ++j) {
+      double d = static_cast< double >(j);
+      vec.push_back(d / nbOfDataPoints);
+   }
+   return vec;
+}
+
+// Create array for y coordinates
+std::vector<double> MakeYCoorVec(int nbOfDataPoints) 
+{
+   std::vector<double> vec;
+   for (int j = 0; j < nbOfDataPoints; ++j) {
+      double d = static_cast< double >(j);
+      vec.push_back(d / (nbOfDataPoints / 10));
+   }
+   return vec;
+}
 
+// Create array for weights
+std::vector<double> MakeWeightVec(int nbOfDataPoints) 
+{
+   std::vector<double> vec;
+   for (int j = 0; j < nbOfDataPoints; ++j) {
+      double d = static_cast< double >(j);
+      vec.push_back(d);
+   }
+   return vec;
+}
+
+// Arrays of coordinates and weights
+std::vector<double> xCoords = MakeXCoorVec(3000000);
+std::vector<double> yCoords = MakeYCoorVec(3000000);
+std::vector<double> weights = MakeWeightVec(3000000);
+
+// Histograms with different bin configurations for Fill
+TH2D hist1{"hist1", "", 10, 0.1, 1., 10 / 2, 0., 10.};
+TH2D hist2{"hist2", "", 10, 0.1, 1., 100 / 2, 0., 10.};
+TH2D hist3{"hist3", "", 10, 0.1, 1., 1000 / 2, 0., 10.};
+TH2D hist4{"hist4", "", 100, 0.1, 1., 10 / 2, 0., 10.};
+TH2D hist5{"hist5", "", 100, 0.1, 1., 100 / 2, 0., 10.};
+TH2D hist6{"hist6", "", 100, 0.1, 1., 1000 / 2, 0., 10.};
+TH2D hist7{"hist7", "", 1000, 0.1, 1., 10 / 2, 0., 10.};
+TH2D hist8{"hist8", "", 1000, 0.1, 1., 100 / 2, 0., 10.};
+TH2D hist9{"hist9", "", 1000, 0.1, 1., 1000 / 2, 0., 10.};
+
+std::map<int, std::map<int, TH2D*>> histFill = { { 10, { { 5, &hist1 }, { 50, &hist2 }, { 500, &hist3} } },
+                                    { 100, { { 5, &hist4 }, { 50, &hist5 }, { 500, &hist6 } } },
+                                    { 1000, { { 5, &hist7 }, { 50, &hist8 }, { 500, &hist9 } } } };
+
+static void FillHist(int nbOfDataPoints, int nbOfBinsForFirstAxis, int nbOfBinsForSecondAxis) 
```

Comment:
I'd have passe the `TH2D` in, instead of the indexes into the map. But that's details.

Reply:
Should I change it?